cascades-server/configuration/src/main/java/pl/gov/coi/cascades/server/persistance/hibernate/HibernateConfiguration.java

```diff
@@ -20,23 +22,37 @@
 public class HibernateConfiguration {
 
     @Bean
-    @Transactional
     TemplateIdGateway createTemplateIdGateway() {
         return new TemplateIdGatewayImpl();
     }
 
     @Bean
-    @Transactional
-    UserGateway createUserGateway(DatabaseTypeClassNameService databaseTypeClassNameService) {
-        return new UserGatewayImpl(databaseTypeClassNameService);
+    UserGateway createUserGateway(UserMapper userMapper) {
+        return new UserGatewayImpl(userMapper);
+    }
+
+    @Bean
+    DatabaseIdGateway createDatabaseIdGateway(DatabaseInstanceMapper databaseInstanceMapper) {
+        return new DatabaseIdGatewayImpl(databaseInstanceMapper);
+    }
+
+    @Bean
+    DatabaseInstanceGateway createDatabaseInstanceGateway(DatabaseInstanceMapper databaseInstanceMapper) {
+        return new DatabaseInstanceGatewayImpl(databaseInstanceMapper);
+    }
+
+    @Bean
+    DatabaseLimitGateway createDatabaseLimitGateway(DatabaseInstanceMapper databaseInstanceMapper) {
+        return new DatabaseLimitGatewayImpl(databaseInstanceMapper);
     }
 
     @Bean
-    @Transactional
-    DatabaseIdGateway createDatabaseIdGateway(DatabaseTypeClassNameService databaseTypeClassNameService) {
-        return new DatabaseIdGatewayImpl(
-            databaseTypeClassNameService
-        );
+    UserMapper createUserMapper(DatabaseTypeClassNameService databaseTypeClassNameService) {
+        return new UserMapper(databaseTypeClassNameService);
     }
 
+    @Bean
+    DatabaseInstanceMapper createDatabaseInstanceMapper(DatabaseTypeClassNameService databaseTypeClassNameService) {
```

Comment:
Maybe this will be unnesesary too

Reply:
Nope.
This method needs this:

 DatabaseIdGateway createDatabaseIdGateway(DatabaseInstanceMapper databaseInstanceMapper) {
        return new DatabaseIdGatewayImpl(databaseInstanceMapper);
 }